libs/nest/audit/src/lib/audit.interceptor.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { of } from 'rxjs'
+import MockInstance = jest.MockInstance
+
+jest.mock('@island.is/auth-nest-tools', () => ({
+  getCurrentUser: jest.fn(),
+}))
+
+const context: any = {
+  getHandler: jest.fn(),
+  getClass: jest.fn(),
+}
+
+const next = {
+  handle: jest.fn(() => of('hello')),
+}
+
+describe('AuditInterceptor', () => {
+  const auditService = mock<AuditService>()
+  let interceptor: AuditInterceptor
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: AuditService,
+          useValue: auditService,
+        },
+        AuditInterceptor,
+      ],
+    }).compile()
+
+    interceptor = module.get<AuditInterceptor>(AuditInterceptor)
+  })
+
+  // Cleanup
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('logs to audit log', async () => {
+    // Arrange
+    class MyClass {
+      @Audit()
+      handler() {}
+    }
+    context.getClass.mockReturnValue(MyClass)
+    context.getHandler.mockReturnValue(MyClass.prototype.handler)
+
+    // Act
+    const observable = interceptor.intercept(context, next)
+    await observable.toPromise()
+
+    // Assert
+    expect(auditService.auditPromise).toHaveBeenCalledWith(
+      expect.objectContaining({
+        action: 'handler',
+      }),
+      expect.anything(),
+    )
+  })
+
+  it('merges content from class decorator', async () => {
+    // Arrange
+    @Audit({ namespace: '@test.is' })
+    class MyClass {
+      @Audit({ action: 'test' })
+      handler() {}
+    }
+    context.getClass.mockReturnValue(MyClass)
+    context.getHandler.mockReturnValue(MyClass.prototype.handler)
+
+    // Act
+    const observable = interceptor.intercept(context, next)
+    await observable.toPromise()
+
+    // Assert
+    expect(auditService.auditPromise).toHaveBeenCalledWith(
+      expect.objectContaining({
+        namespace: '@test.is',
+        action: 'test',
+      }),
+      expect.anything(),
+    )
+  })
+
+  it('forwards next result', async () => {
+    // Arrange
+    class MyClass {
+      @Audit()
+      handler() {}
+    }
+    context.getClass.mockReturnValue(MyClass)
+    context.getHandler.mockReturnValue(MyClass.prototype.handler)
+    const result = 'testResult'
+    next.handle.mockReturnValue(of(result))
+
+    // Act
+    const observable = interceptor.intercept(context, next)
+    await observable.toPromise()
+
+    // Assert
+    expect(auditService.auditPromise).toHaveBeenCalledWith(
+      expect.anything(),
+      expect.any(Promise),
+    )
+    await expect(auditService.auditPromise.mock.calls[0][1]).resolves.toEqual(
+      result,
+    )
+  })
+
+  it('forwards user', async () => {
+    // Arrange
+    class MyClass {
+      @Audit()
+      handler() {}
+    }
+    context.getClass.mockReturnValue(MyClass)
+    context.getHandler.mockReturnValue(MyClass.prototype.handler)
+    const user = 'Test user'
+    ;((getCurrentUser as unknown) as MockInstance<
```

Comment:
Is this: https://prettier.io/docs/en/rationale.html#semicolons what's happening here?

Reply:
Yeah... :( Will rewrite as variable.